dashboard/src/components/OperatorNew/OperatorNew.test.tsx

```diff
@@ -0,0 +1,124 @@
+          annotations: {
+            capabilities: "Basic Install",
+            repository: "github.com/kubeapps/kubeapps",
+            containerImage: "kubeapps/kubeapps",
+            createdAt: "one day",
+          },
+          installModes: [],
+        },
+      },
+    ],
+  },
+} as any;
+
+it("calls getOperator when mounting the component", () => {
+  const getOperator = jest.fn();
+  shallow(<OperatorNew {...defaultProps} getOperator={getOperator} />);
+  expect(getOperator).toHaveBeenCalledWith(defaultProps.namespace, defaultProps.operatorName);
+});
+
+it("calls getOperator when changing the namespace the component", () => {
+  const getOperator = jest.fn();
+  const wrapper = shallow(<OperatorNew {...defaultProps} getOperator={getOperator} />);
+  expect(getOperator).toHaveBeenCalledWith(defaultProps.namespace, defaultProps.operatorName);
+  wrapper.setProps({ namespace: "foo" });
+  expect(getOperator).toHaveBeenCalledWith("foo", defaultProps.operatorName);
+});
+
+it("parses the default channel when receiving the operator", () => {
+  const wrapper = shallow(<OperatorNew {...defaultProps} />);
+  wrapper.setProps({ operator: defaultOperator });
+  expect(wrapper.state()).toMatchObject({
+    updateChannel: defaultOperator.status.channels[0],
+    updateChannelGlobal: false,
+    installationModeGlobal: false,
+  });
+});
+
+it("shows an error if it exists", () => {
+  const wrapper = shallow(<OperatorNew {...defaultProps} error={new NotFoundError()} />);
+  expect(wrapper.html()).toContain("Operator foo not found");
+});
+
+it("shows an error if the operator doesn't have any channel defined", () => {
+  const operator = {
+    status: {
+      channels: [],
+    },
+  };
+  const wrapper = shallow(<OperatorNew {...defaultProps} operator={operator as any} />);
+  expect(
+    wrapper
+      .find(UnexpectedErrorPage)
+      .dive()
+      .text(),
+  ).toContain(
+    "Operator foo doesn't define a valid channel. This is needed to extract required info",
+  );
+});
+
+it("renders a full OperatorNew", () => {
+  const wrapper = shallow(<OperatorNew {...defaultProps} />);
+  wrapper.setProps({ operator: defaultOperator });
+  expect(wrapper).toMatchSnapshot();
+});
+
+it("disables the submit button if the operators ns is selected", () => {
+  const wrapper = shallow(<OperatorNew {...defaultProps} namespace="operators" />);
+  wrapper.setProps({ operator: defaultOperator });
+  expect(wrapper.find("button").prop("disabled")).toBe(true);
+  expect(wrapper.find(UnexpectedErrorPage)).toExist();
+});
+
+it("deploys an operator", async () => {
+  const createOperator = jest.fn(() => true);
+  const push = jest.fn();
+  const wrapper = shallow(
+    <OperatorNew
+      {...defaultProps}
+      namespace="operators"
+      createOperator={createOperator}
+      push={push}
+    />,
+  );
+  wrapper.setProps({ operator: defaultOperator });
+  wrapper.find("form").simulate("submit");
+  await wait(1);
```

Comment:
hmph... did you try some of the things I mentioned last time? Are you certain we're not adding potential flaky tests which are going to bite us in the future? I'd much prefer that we not continue sprinkling the tests with wait, and instead (in order of my preference, but I don't care):
1) test the onSubmit handler separately - we don't need to test that simulating the form's submit event triggers the handler, necessarily,
2) using mount instead of shallow, and simulating on the button instead (see https://github.com/enzymejs/enzyme/issues/308 for other ideas, though I've not tested them).